src/hotspot/share/gc/shared/collectedHeap.cpp

```diff
@@ -548,8 +548,8 @@ void CollectedHeap::full_gc_dump(GCTimer* timer, bool before) {
     GCTraceTime(Trace, gc, classhisto) tm(before ? "Class Histogram (before full gc)" : "Class Histogram (after full gc)", timer);
     ResourceMark rm;
     LogStream ls(lt);
-    uint parallel_thread_num = MAX2<uint>(1, (uint)os::initial_active_processor_count() * 3 / 8);
-    VM_GC_HeapInspection inspector(&ls, false /* ! full gc */, parallel_thread_num);
+    uint parallel_thread_num = safepoint_workers()->active_workers();
+    VM_GC_HeapInspection inspector(&ls, false /* request full gc */, parallel_thread_num);
```

Comment:
I would prefer the same issue would also be fixed in the other place (`ClassHistogramDCmd::execute`), using a static method as suggested. Otherwise the problem will exist forever.